api/storage_service.cc

```diff
@@ -510,6 +510,54 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         });
     });
 
+    ctx.db.local().get_builtin_routine_registry().register_routine("toppartitions",
+        db::make_single_value_routine<sstring>("toppartitions", [&ctx] (db::builtin_routine::parameter_map&& par) {
+            bool filters_provided = false;
+
+            std::unordered_set<std::tuple<sstring, sstring>, utils::tuple_hash> table_filters {};
+            if (par.contains("table_filters")) {
+                filters_provided = true;
+                auto filters = par["table_filters"];
+                std::stringstream ss { filters };
+                std::string filter;
+                while (!filters.empty() && ss.good()) {
```

Comment:
`!filters.empty()` is not a valid condition

Reply:
what do you mean?

Reply:
I mean, it's not wrong, but `filters` does not change inside the loop and such a condition prevents it from being `std::move`d into `stringstream`. If anything, it might make sense to check for emptiness of `filter` (if it read any data, but that should be done before attempting to emplace it into `table_filters`).
Anyway, that should be sufficient:
```
while (std::getline(ss, filter, ',')) {
```